pkgs/by-name/no/nordvpn/package.nix

```diff
@@ -0,0 +1,150 @@
+        patches =
+          (old.patches or [ ])
+          ++ (
+            let
+              patchedVersion = "2.6.14";
+              patchURL = "${baseURL}/openvpn-${patchedVersion}/patches";
+            in
+            [
+              (prev.fetchpatch {
+                url = "${patchURL}/02-tunnelblick-openvpn_xorpatch-a.diff";
+                hash = "sha256-b9NiWETc0g2a7FNwrLaNrWx7gfCql7VTbewFu3QluFk=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/03-tunnelblick-openvpn_xorpatch-b.diff";
+                hash = "sha256-X/SshB/8ItLFBx6TPhjBwyA97ra0iM2KgsGqGIy2s9I=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/04-tunnelblick-openvpn_xorpatch-c.diff";
+                hash = "sha256-fw0CxJGIFEydIVRVouTlD1n275eQcbejUdhrU1JAx7g=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/05-tunnelblick-openvpn_xorpatch-d.diff";
+                hash = "sha256-NLRtoRVz+4hQcElyz4elCAv9l1vp4Yb3/VJef+L/FZo=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/06-tunnelblick-openvpn_xorpatch-e.diff";
+                hash = "sha256-mybdjCIT9b6ukbGWYvbr74fKtcncCtTvS5xSVf92T6Y=";
+              })
+            ]
+          );
+      });
+    })
+  ];
+in
+buildGoModule rec {
+  pname = "nordvpn";
+  version = "3.20.2";
+
+  src = fetchFromGitHub {
+    owner = "NordSecurity";
+    repo = "nordvpn-linux";
+    rev = version;
+    sha256 = "sha256-XB/wOdp97VnAkzHTHrJ2xffEx4HNFWn9yjFz3TsaEsE=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    gcc
+  ];
+
+  buildInputs = [
+    cacert
+    gcc
+    iproute2
+    iptables
+    libidn2
+    libxml2
+    procps
+    sysctl
+    wireguard-tools
+    zlib
+  ];
+
+  vendorHash = "sha256-gW/KieixLS6TTcBOHeYK9L8sieERrj6ia+iov8rT2AE=";
+
+  ldflags = [
+    "-X main.Version=${version}"
+    "-X main.Environment=dev"
+    "-X main.Salt=development"
+    "-X main.Hash=${src.rev}"
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    echo "Building nordvpn CLI..."
+    sed -i "s|/usr/lib/nordvpn|$out/bin|g" internal/constants.go
+
+    export LDFLAGS="${builtins.concatStringsSep " " ldflags}"
+    go build -ldflags "$LDFLAGS" -o bin/nordvpn ./cmd/cli
+
+    echo "Building nordvpn user..."
+    go build -ldflags "$LDFLAGS" -o bin/norduserd ./cmd/norduser
+
+    # Fix missing include in a library preventing compilation
+    chmod +w vendor/github.com/jbowtie/gokogiri/xpath/
+    sed -i '6i#include <stdlib.h>' vendor/github.com/jbowtie/gokogiri/xpath/expression.go
+
+    echo "Building nordvpn daemon..."
+    go build -ldflags "$LDFLAGS" -o bin/nordvpnd ./cmd/daemon
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    BIN_PATH=$out/bin
+    mkdir -p $BIN_PATH
+    mv bin/norduserd $BIN_PATH
+    ln -s ${patchedPkgs.openvpn}/bin/openvpn $BIN_PATH
+    ln -s ${wireguard-tools}/bin/wg $BIN_PATH
```

Comment:
This leads to leaking the patched openvpn and wireguard-tools into the bin directory for the end user.
Can we patch nordvpn to give them a full path to the binary directly?